src/pastel/winConsoleColorsSupport.c

```diff
@@ -28,21 +31,19 @@ CAMLprim value enable_windows_console_ansi_sequences()
   int success = 0;
 
   HANDLE hOut = GetStdHandle(STD_OUTPUT_HANDLE);
-
-  if (hOut != INVALID_HANDLE_VALUE)
+  DWORD dwMode = 0;
+  // GetConsoleMode returns 6 "invalid handle" when running in cygwin build /
+  // dev shell. You don't need to enable ansi sequences anyways in that case.
+  // So if GetConsoleMode returns false, just give up.
+  if (hOut != INVALID_HANDLE_VALUE && GetConsoleMode(hOut, &dwMode))
   {
-      DWORD dwMode = 0;
-
-      if (GetConsoleMode(hOut, &dwMode))
-      {
-          dwMode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;
-          success = SetConsoleMode(hOut, dwMode) ? 1 : 0;
-      }
+    dwMode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;
+    success = SetConsoleMode(hOut, dwMode) ? 1 : 0;
+    if (success != 1) {
+      caml_failwith( "error in enable_windows_console_ansi_seqences");
```

Comment:
Can this be a `return Val_int(-1)` instead indicating the `SetConsoleMode` failed?

Reply:
If you can get the console mode but not set it, then I think that is a big issue and we should hard fail as it does here. But before this diff, we couldn't get the console mode, nor set it - and that's because if you are running in a mintty environment, you cannot do either. Let's see if this works, and if not I'll probably have to come back and apply your suggestion.